### DISCUSSION TITLE
treeTable.getData疑问

### DISCUSSION THREAD

**Discussion by hjingsuper** [layui/layui#1762] [讨论]
示例：[https://codepen.io/hjingsuper/pen/NWmyoae](https://codepen.io/hjingsuper/pen/NWmyoae)

> 1.获取树表数据 treeTable.getData(id, isSimpleData); 会返回内部的特定字段

文档表示：不包含 treeTable 组件内部的特定字段。
![image](https://github.com/layui/layui/assets/36070786/aa5a0ac0-4699-4d5d-a38a-f4483c870312)
理论上只要返回原始数据和customName中包含的字段

table就没有包含特定字段，以下是table.getData(id);
![image](https://github.com/layui/layui/assets/36070786/8564c48c-b930-4c22-b5f8-53d1ddf5109a)

> 2.获取树表数据 treeTable.getData(id, isSimpleData); 第二个参数获取不明确，简单数据结构在treeTable要怎么理解

文档表示：参数isSimpleData是否为简单数据，为 true 时返回简单数据结构的数据，否则则为带层级的数据
按照结果来看，为 true 时返回每一行的数据，同时也会带有children层级。这样看起来好像isSimpleData参数用处不大
![image](https://github.com/layui/layui/assets/36070786/0a2d97bf-b664-407d-9998-40e3aae1042b)

> 3.treeTable.checkStatus(id);同上

看结果应该是放回的层级数据结构，同样会包含特定字段

**想法**
`treeTable.getData(id, true)` 返回每一行原始数据，不包括children和特定字段
`treeTable.getData(id)` 带children层级的数据，不包括特定字段
`treeTable.checkStatus(id)` 不包括特定字段

**Comment by Sight-wcg** [comment]
> 1.获取树表数据 treeTable.getData(id, isSimpleData); 会返回内部的特定字段

treeTable.getData 返回的数据确实和文档有一定出入，从源码来看，去掉返回数据中的 LAY_XXX 特定字段，对内部功能会有影响。可能指的是像 ` 0-0, 0-1` 这样临时的 key
比如原始数据为
```
data: [
  {id:0, children:[id:1, pid: 0]},
  {id:2, children:[id,pid: 2]}
]
```
treeTable 将其作为关联数组处理, `0-0,1-0` 是为了方便访问数据生成的临时 key，引用根节点各个层级的子节点
```
{
  0 :{id:0, children:[{id:1, pid: 0]}},
  0-0: {id: 1},
  1: {id:2, children:[{id:3,pid: 2}]}
  1-0: {id:3},
  length: 2
}
```

2，3 后续版本会考虑改进。

有劳 @sentsim @sunxiaobin89 确认一下这个问题。

**Reply by hjingsuper** [reply]
好的，谢谢答疑

**Reply by hjingsuper** [reply]
> 1.获取树表数据 treeTable.getData(id, isSimpleData); 会返回内部的特定字段

_**去掉返回数据中的 LAY_XXX 特定字段，对内部功能会有影响。**_ 
    
特定字段确实不能去掉，treeTable多样化编辑，修改其他字段值需要依靠LAY_DATA_INDEX实现。还是保留现状比较好
示例：[https://codepen.io/hjingsuper/pen/MWRGrrr](https://codepen.io/hjingsuper/pen/MWRGrrr)

```
      var tableViewElem = this.elem.next();
      // 解除 tbSelect 命名空间下的所有 change 事件处理程序
      tableViewElem.off("change.tbSelect");
      // 将 '.select-sex' 元素的 change 事件委托给 tableViewElem, 事件命名空间为 tbSelect
      tableViewElem.on("change.tbSelect", ".select-sex", function () {
        var value = this.value; // 获取选中项 value
        var data = table.getRowData(options.id, this);
        data.sex = value;

        var DATA_INDEX = data.LAY_DATA_INDEX; //此处需要内部字段更新行数据
        treeTable.updateNode("MianTable", DATA_INDEX, {
          province: "01"
        });
      });
```
